### ISSUE TITLE
Create a visual representation of our Open Source workflow

### ISSUE THREAD

**Issue by isabelcosta** [anitab-org/open-source-teams#3]
## Problem

As we evolve as a community, we are evolving our open source workflow. Currently we have been trying to add a QA testing step to how we handle pull requests, and this is not yet visually represented. I think this could clear up confusion from new and current contributors.

In the past, we had this open source workflow: http://systers.io/open-source-workflow

Today we have a specific way of assigning labels to define the status of PRs and issues. We want to have this visually represented.

To help understand some of our workflow currently being followed, check our [Open Source teams guide](https://docs.google.com/document/d/1bt1Dfi_LK8-IDOX-vxgc_PT-CGP4sWhSI5TT4amtLMY/edit).

## Tasks

- [ ] Create a visual representaiton that explains ourcurrent open source wokflow.

**Comment by mitchell-frost** [comment]
I am new here. Can I work on this to get started?

**Comment by isabelcosta** [comment]
@mitchell-frost sure you can :) Do you understand what this is about and where you can find information to build this representation?

**Comment by mitchell-frost** [comment]
@isabelcosta I am actually not sure where to start. Any help shall be appreciated!

**Comment by isabelcosta** [comment]
@mitchell-frost no problem, I would start by looking at the document I mention in the issue description - [Open Source teams guide](https://docs.google.com/document/d/1bt1Dfi_LK8-IDOX-vxgc_PT-CGP4sWhSI5TT4amtLMY/edit) - so you understand the label system we use. 

For example, these are some of the things we'd like to see in the representation:
workflow since someone creates an issue until a PR of that issue is merged. We use some labels to manage the PRs current status. For example, after we have 2 minimum code reviews we add the "Status: needs Testing" label. After at least someone reports a tests done to the PR, if no issue arises, we add label "Status: Ready to Merge" so that maintainers know they can merge the PR.

You can also see a past workflow representation in the issue description. Do you understand this is not a coding issue but a documentation one? The result of this may get into our https://anitab-org.github.io website, and general documentation in this repository. This is something to guide our current and future contributors.

Are you on our Zulip? If not i would advise you to join, you may understand some of our workflow discussions there.

👉  https://anitab-org.zulipchat.com/

**Comment by mitchell-frost** [comment]
@isabelcosta thanks! I will start working on it and I understand that it's a documentation issue.

**Comment by isabelcosta** [comment]
@mitchell-frost  Feel free to use GitHub comments or Zulip messages or an office hour of mine to discuss this and ask more questions. I am having an office hour today at 23h (Lisbon timezone) :)

**Comment by isabelcosta** [comment]
@mitchell-frost Any updates? Do you need any help?

**Comment by mitchell-frost** [comment]
I am working on another issue as well. I will keep you updated about the progress @isabelcosta

**Comment by isabelcosta** [comment]
@mitchell-frost its better to focus on one issue at a time. Do you mind if I make this available again, until you are available again? This is a high priority issue, because there is a lot of confusion in the community with what we are doing in the community

**Comment by nandini45** [comment]
@isabelcosta  she is already working on a issue in anitab-org.github.io open source website (https://github.com/anitab-org/anitab-org.github.io/issues/42). 
its better to make this issue available again.
and i agree we do need to setup the contribution guidelines again for all the repository especially for newcomers and other contributions

**Comment by gaurivn** [comment]
@mitchell-frost , which issue are you working on atm, is it this one or the other one that @nandini45 mentioned in previous message? Please tell which one you're working on currently, so we can make the other one available as @isabelcosta said.

**Comment by isabelcosta** [comment]
@nandini45 thank you for the update, I will make this available again :)

**Comment by rpattath** [comment]
Workflow requirements:

When the PR is in `Status: Needs Review` - 2 minimum approved code reviews.

When the PR is in `Status: Needs Testing`: minimum 1 tester has to test locally and [report](https://github.com/anitab-org/mentorship-backend/blob/develop/docs/test-pr-guide.md#template-to-report-pr-testing-results) it as a comment on the PR.

Tester says:
→ success → “Status: Ready to Merge”
→ failure → “Status: Needs Review”

**Comment by mitchell-frost** [comment]
@isabelcosta @nandini45 Apologies about the confusion. I am new here. I will try not to take up two issues at once.

**Comment by isabelcosta** [comment]
@mitchell-frost no worries at all! We prefer for contributors to work on one issue at a time so that others can take on other issues. Now you know :) I hope you are enjoying working on the other issue.

**Comment by mayburgos** [comment]
@rpattath I can work on this. :)

**Comment by rpattath** [comment]
Thank you @mayburgos ! Let me know if I could be of any help :)

**Comment by isabelcosta** [comment]
@rpattath I talked about this in the Content Open Session today, and @mayburgos agreed to help you with the visual representation, your PR is still going to be part of this work :)

**Comment by rpattath** [comment]
@isabelcosta sounds like a great plan !

@mayburgos let me know if you want me to come up with a visual representation for you to review, or if you are coming up with one I would be more than happy to look at it.

**Comment by rpattath** [comment]
@mayburgos @isabelcosta please release me from this Issue. I think integration test automation is a very pressing issue for qa-team as of now. I would like to focus on that. I am sorry.

**Comment by isabelcosta** [comment]
@rpattath no need to apologize. We completely understand, and you updated us, that's all we need :)

**Comment by noobkid2411** [comment]
Hi,this is Sonakshi. I am new here and would very much like to contribute to this issue.I hope I can help in resolving it. Can I get started on this? Any help would be appreciated.

**Comment by devifeoma** [comment]
Hi @isabelcosta, I’m Judith and I would like to contribute to this.  I await your response.

**Comment by isabelcosta** [comment]
@noobkid2411 are you still interested in working on this? If yes, I'll assign to you if not, no problem, I'll assign to @devifeoma :)

**Comment by devifeoma** [comment]
Hi @noobkid2411 <https://github.com/noobkid2411>, I await your swift
response.

**Comment by noobkid2411** [comment]
@isabelcosta @devifeoma yeah I'm still interested ,I'd like to start on this issue.

**Comment by isabelcosta** [comment]
cool will assign it to you @noobkid2411 :)

**Comment by isabelcosta** [comment]
@devifeoma we have another project with a documentation issue, if you'd like to work on it, leave a comment and I will assign it to you :)
Issue: https://github.com/anitab-org/open-source-programs-web/issues/71

**Comment by devifeoma** [comment]
Hi Isabel, yes I’m interested in working project. Let me know about it.

**Comment by devifeoma** [comment]
Yes, @isabelcosta <https://github.com/isabelcosta> assign to me.
